tests/unittest_brain.py

```diff
@@ -1266,60 +1439,111 @@ class Derived2(typing.OrderedDict[int, str]):
+                "Hashable",
                 "object",
             ],
         )
+        with self.assertRaises(astroid.exceptions.AttributeInferenceError):
+            inferred.getattr("__class_getitem__")
 
-    @test_utils.require_version("3.8")
-    def test_typing_alias_side_effects(self):
-        """Test that typing._alias changes doesn't have unwanted consequences."""
-        node = builder.extract_node(
+    @test_utils.require_version(minver="3.7")
+    def test_typing_object_subscriptable(self):
+        """Test that MutableSet is subscriptable"""
+        right_node = builder.extract_node(
             """
         import typing
-        import collections.abc
 
-        class Derived(collections.abc.Iterator[int]):
-            pass
+        typing.MutableSet[int]
         """
         )
-        inferred = next(node.infer())
-        assert inferred.metaclass() is None  # Should this be ABCMeta?
+        inferred = next(right_node.infer())
+        check_metaclass_is_abc(inferred)
         assertEqualMro(
             inferred,
             [
-                "Derived",
-                # Should this be more?
-                # "Iterator_typing"?
-                # "Iterator",
-                # "object",
+                "MutableSet",
+                "Set",
+                "Collection",
+                "Sized",
+                "Iterable",
+                "Container",
+                "object",
             ],
         )
+        self.assertIsInstance(
+            inferred.getattr("__class_getitem__")[0], nodes.FunctionDef
+        )
+
+    @test_utils.require_version(minver="3.7")
+    def test_typing_object_notsubscriptable_3(self):
+        """Until python39 ByteString class of the typing module is not subscritable (whereas it is in the collections module)"""
+        right_node = builder.extract_node(
+            """
+        import typing
+
+        typing.ByteString
+        """
+        )
+        inferred = next(right_node.infer())
+        check_metaclass_is_abc(inferred)
+        with self.assertRaises(astroid.exceptions.AttributeInferenceError):
+            self.assertIsInstance(
+                inferred.getattr("__class_getitem__")[0], nodes.FunctionDef
+            )
+
+    @test_utils.require_version(minver="3.9")
+    def test_typing_object_builtin_subscriptable(self):
+        """
+        Test that builtins alias, such as typing.List, are subscriptable
+        """
+        #  Do not test Tuple as it is inferred as _TupleType class (needs a brain?)
+        for typename in ("List", "Dict", "Set", "FrozenSet"):
+            src = """
+            import typing
+            typing.{:s}[int]
+            """.format(
+                typename
+            )
+            right_node = builder.extract_node(src)
+            inferred = next(right_node.infer())
+            self.assertIsInstance(inferred, ClassDef)
+            self.assertIsInstance(inferred.getattr("__iter__")[0], FunctionDef)
```

Comment:
`ClassDef` and `FunctionDef` are also accessible through `nodes.ClassDef` and `nodes.FunctionDef`. That would be more consistent with your other tests.